jobs/mysql_backup_job.go

```diff
@@ -146,43 +173,42 @@ func (j *MySQLBackupJob) Run(ctx context.Context, send Send) (rerr error) {
 			errBackupBuffer.String(), outBuffer.String(), errCloudBuffer.String())
 	}
 
+	if err := xtrabackupCmd.Start(); err != nil {
+		cancel()
+		return wrapError(err)
+	}
+
+	defer func() {
+		if err := xtrabackupCmd.Wait(); err != nil {
+			cancel()
+			if rerr != nil {
+				rerr = errors.Wrapf(rerr, "xtrabackup wait error: %s", err)
+			} else {
+				rerr = wrapError(err)
+			}
+		}
+	}()
+
 	if xbcloudCmd != nil {
 		xbcloudCmd.Stdin = xtrabackupStdout
 		xbcloudCmd.Stdout = &outBuffer
 		xbcloudCmd.Stderr = &errCloudBuffer
 		if err := xbcloudCmd.Start(); err != nil {
+			cancel()
```

Comment:
the same here. Because firstly we start both process and just afterwards we call `Wait()` so this doesn't seems necessary?

Reply:
Here it's required, because if we failed to start `xbcloud`  we invoke return, and that triggers defer with `xtrabackupCmd.Wait()`  where we will stuck.